src/shared_modules/indexer_connector/src/monitoring.hpp

```diff
@@ -90,87 +168,32 @@ class Monitoring final
      */
     explicit Monitoring(const std::vector<std::string>& serverAddresses,
                         const uint32_t interval = INTERVAL,
                         const SecureCommunication& secureCommunication = {})
         : m_interval(interval)
     {
-        /**
-         * @brief Health check function
-         *
-         * @note It sends a request to the \p serverAddress and update the \p serverStatus (true if the server is
-         * available, false otherwise).
-         */
-        const auto healthCheck = [this, secureCommunication](const std::string& serverAddress, bool& serverStatus)
-        {
-            // Get the health of the server.
-            HTTPRequest::instance().get(
-                HttpURL(serverAddress + "/_cat/health?v"),
-                [&](std::string response)
-                {
-                    // Remove the tabs and double spaces.
-                    Utils::replaceAll(response, "\t", " ");
-                    response = Utils::trimRepeated(response, ' ');
-                    // Split the response by rows.
-                    const auto rows {Utils::split(response, '\n')};
-
-                    // Check if the response has the expected number of rows.
-                    if (HealthCheckRows::SIZE == rows.size())
-                    {
-                        // Split the data row by spaces.
-                        const auto fields {Utils::split(rows.at(HealthCheckRows::DATA), ' ')};
-
-                        // Check if the response has the expected number of columns and if the status is
-                        // green.
-                        if (fields.size() == HealthCheckColumns::SIZE &&
-                            fields.at(HealthCheckColumns::STATUS).compare("green") == 0)
-                        {
-                            serverStatus = true;
-                        }
-                        else
-                        {
-                            serverStatus = false;
-                        }
-                    }
-                    else
-                    {
-                        serverStatus = false; // LCOV_EXCL_LINE
-                    }
-                },
-                [&](const std::string& error, const long /*statusCode*/) { serverStatus = false; },
-                "",
-                {"Accept-Charset: utf-8"},
-                secureCommunication);
-        };
-
-        {
-            // Initialize the status of the servers
-            std::scoped_lock lock(m_mutex);
-            for (const auto& serverAddress : serverAddresses)
-            {
-                healthCheck(serverAddress, m_servers[serverAddress]);
-            }
-        }
+        initialize(serverAddresses, secureCommunication);
 
         // Start the thread, that will check the health of the servers.
         m_thread = std::thread(
-            [this, secureCommunication, healthCheck]()
+            [this, &secureCommunication]()
```

Comment:
warning with this, maintain secureCommunication as copy, because the lifetime of this variable is only guaranted by the IndexerConnector constructor.

IndexerConnector() local variable -> ServerSelector() pass by reference -> Monitoring() pass by reference -> std::thread pass by reference.

When the IndexerConnector constructor returns, this variable is destroyed.